src/main/java/tornadofx/Layouts.kt

```diff
@@ -33,6 +33,43 @@ fun GridPane.row(title: String? = null, op: Pane.() -> Unit = {}) {
     addRow(properties[GridPaneRowIdKey] as Int, *fake.children.toTypedArray())
 }
 
+/**
+ * Removes the corresponding row to which this [node] belongs to.
+ *
+ * It does the opposite of the [GridPane.row] cleaning all internal state properly.
+ *
+ * @return the row index of the removed row.
+ */
+fun GridPane.removeRow(node: Node): Int {
+    if (properties.containsKey(GridPaneRowIdKey)) {
+        properties[GridPaneRowIdKey] = properties[GridPaneRowIdKey] as Int - 1
```

Comment:
Here the alternative implementation could be if the property value is 0, just to remove the property all together instead of subtract.

Although currently it doesn't matter, as in the current case it will result in `-1`, and when adding the new row the index will become `0`, same as if the property would absent.